### DISCUSSION TITLE
Support for ACF Fields

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/publishpress-checklists#429] [Feature Requests]
Possible Pro feature: support for ACF

> It would be great to check if a field is filleed in. We could specify the name of a specific ACF field that should be field in, and you could use for this maybe data-name="act-field-name" which corresponds to the ACF filed name.

> Post editor page:

![Screenshot-2020-02-19-at-16-42-16](https://user-images.githubusercontent.com/3868028/74856180-9ec25a80-5339-11ea-9127-75f335ddefdf.png)

> ACF field group page with the name of the field that someone might want to check if it was filled:

<img width="640" alt="Screenshot-2020-02-19-at-16-43-32" src="https://user-images.githubusercontent.com/3868028/74856200-a4b83b80-5339-11ea-811e-adba700cf865.png">

**Comment by ghost** [comment]
We'd like to see this feature added as well. It would be great if the Checklist plugin supported validation against Advanced Custom Fields. Our scenario is we want a few custom fields completed before a Post/news release is published. The best we can do is add a recommendation:

![image](https://user-images.githubusercontent.com/26336976/89594830-c766df80-d807-11ea-8523-9877fdff9eb4.png)

**Comment by stevejburge** [comment]
Thanks @webmaster-cob. I've been thinking about how to approach ACF.

Presumably we need to automatically check for all available ACF fields and then offer them as checklist options.

It becomes more complex because of the all the possible fields types: https://www.advancedcustomfields.com/resources/#field-types

Maybe we start by just checking that each one has content?

**Comment by ghost** [comment]
That would be a great start ;)

**Comment by stevejburge** [comment]
One more request:

> We have a few ACF fields like image, description and some select list which an author should fill in before they send the post for approval, if they don't fill in those fields then they should be thrown the error message, once they fill the fields then only they should be allowed to submit the post for approval.

**Comment by CodingSamurai** [comment]
Another vote for being able to have checklist actions around ACF fields.

**Comment by stevejburge** [comment]
Now shipped

**Comment by CodingSamurai** [comment]
@stevejburge what version of the Checklist did this roll out in?

**Comment by stevejburge** [comment]
2.14.0
